code/dictation.py

```diff
@@ -61,8 +88,9 @@ def capture_to_words(m):
   )$""", re.VERBOSE)
 no_space_before = re.compile(r"""
   ^(?:
-    [\s\-_.,!?;:/%)\]}’”]   # characters that never need space before them
-  | [$£€¥₩₽₹](?!\w)         # currency symbols not followed by a word character
+    [\s\-_.,!?/%)\]}'’”]   # characters that never need space before them
```

Comment:
I added "apostrophe ess": "'s" to keys.py and that only formats correctly if I make this change. How do you do achieve this?

Reply:
Ah, I don't have that feature, but I can see how it's useful. The reason I'm not sure about making `'` unconditionally require no space before it is because sometimes it's used for quotation instead of `"` (eg. sometimes in British English this is the preferred style). Perhaps we should special-case possessives by adding `'s\b` to no_space_before. Does that seem like a good approach?

Reply:
Sounds good to me; I just made the change.

Reply:
I did slightly different syntax (with style more in keeping with the other regexes here) but the effect should be equivalent.